src/Kovan-DssSpell.t.sol

```diff
@@ -594,70 +596,185 @@ contract DssSpellTest is DSTest, DSMath {
-        assertEq(vat.dai(address(this)), 0);
-
-        // Withdraw from adapter
-        joinGUSDA.exit(address(this), faucetAmount);
-        assertEq(gusd.balanceOf(address(this)), faucetAmount);
-        assertEq(vat.gem("GUSD-A", address(this)), 0);
-
-        // Generate new DAI to force a liquidation
-        gusd.approve(address(joinGUSDA), faucetAmount);
-        joinGUSDA.join(address(this), faucetAmount);
-        (,,uint256 spotV,,) = vat.ilks("GUSD-A");
-        // dart max amount of DAI
-        vat.frob("GUSD-A", address(this), address(this), address(this), int(faucetAmountWad), int(mul(faucetAmountWad, spotV) / RAY));
-        hevm.warp(now + 1);
-        jug.drip("GUSD-A");
-        // assertEq(flipGUSDA.kicks(), 0);
-        // cat.bite("GUSD-A", address(this));
-        // assertEq(flipGUSDA.kicks(), 1);
+        _runOldChief();
     }
 
+    function testMoms() public {
+        vote();
+        scheduleWaitAndCast();
+        assertTrue(spell.done());
+
+        DSTokenAbstract(oldChief.IOU()).approve(address(oldChief), uint256(-1));
+        oldChief.free(999999999999 ether);
+        gov.approve(address(newChief), uint256(-1));
+
+        newChief.lock(80_000 ether);
+        address[] memory slate = new address[](1);
+
+        // Create spell for testing
+        SpellTestMoms spellTestMoms = new SpellTestMoms();
+
+        // System not launched, lifted address doesn't get root access
+        slate[0] = address(spellTestMoms);
+        newChief.vote(slate);
+        newChief.lift(address(spellTestMoms));
+        assertTrue(!newChief.isUserRoot(address(spellTestMoms)));
+
+        // Launch system
+        slate[0] = address(0);
+        newChief.vote(slate);
+        newChief.lift(address(0));
+        newChief.launch();
+
+        // System launched, lifted address gets root access
+        slate[0] = address(spellTestMoms);
+        newChief.vote(slate);
+        newChief.lift(address(spellTestMoms));
+        assertTrue(newChief.isUserRoot(address(spellTestMoms)));
+
+        FlipAbstract flip = FlipAbstract(changelog.getAddress("MCD_FLIP_ETH_A"));
+        OsmAbstract osm   = OsmAbstract(changelog.getAddress("PIP_ETH"));
+
+        assertEq(flip.wards(address(cat)), 1);
+        assertEq(osm.stopped(), 0);
+        spellTestMoms.cast();
+        assertEq(flip.wards(address(cat)), 0);
+        assertEq(osm.stopped(), 1);
+    }
+
+    function testSAIcontractsAuthorityChange() public {
+        assertEq(saiMom.authority(), address(oldChief));
+        assertEq(saiTop.authority(), address(oldChief));
+        vote();
+        spell.schedule();
+        assertEq(saiMom.authority(), address(0));
+        assertEq(saiTop.authority(), address(0));
+    }
+}
+
+contract SpellActionTest {
+    function execute() external {
+        // Random action to test authority
+        VatAbstract(ChainlogAbstract(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F).getAddress("MCD_VAT")).rely(address(123));
+    }
+}
+
+contract SpellTest {
```

Comment:
Rather than `SpellTest`, I'd call it `TestSpell` (the former is a little confusing since in our naming conventions it sounds like a contract that contains tests, rather than a mock contract created for testing).

Reply:
Agreed, changing it.